### DISCUSSION TITLE
Compile error when changing time sync source to homeassistant

### DISCUSSION THREAD

**Discussion by Tecratal** [emporia-vue-local/esphome#20] [Q&A] [answered]
SNTP is off by about 3 and a half hours for me for the daily reset. I want to set it to sync to home assistant to avoid any time sync issues using:

time:
  - platform: homeassistant
    id: homeassistant_time

When I run esphome it gives me the error: 

> Compiling .pioenvs\vue1\src\esphome\components\ota\ota_backend_esp_idf.o
> Compiling .pioenvs\vue1\src\esphome\components\ota\ota_component.o
> Compiling .pioenvs\vue1\src\esphome\components\sensor\automation.o
> Compiling .pioenvs\vue1\src\esphome\components\sensor\filter.o
> Compiling .pioenvs\vue1\src\esphome\components\sensor\sensor.o
> *** [.pioenvs\vue1\src\esphome\components\sntp\sntp_component.o] Source `src\esphome\components\sntp\sntp_component.cpp' not found, needed by target `.pioenvs\vue1\src\esphome\components\sntp\sntp_component.o'.
> ========================= [FAILED] Took 10.02 seconds =========================

I am using the same time configuration with sonoff s31 and do not receive any errors during compiling.

**Comment by flaviut** [comment, marked as answer]
Try  "Clean Build Files" then try again:

![87b777643ac9c33c2d6e9b5e73522fa33898dd56](https://user-images.githubusercontent.com/5213469/186049019-12cfe3a8-a27e-4575-aa8b-f01c28160b35.jpeg)

 There's a chance it's a time zone issue, ESPHome gets the default timezone from the environment, which might not be right if it's a VM, etc.

**Reply by Tecratal** [reply]
The clean build files work for me. In case anyone else reads this post I was using it from command line and ran:

esphome clean vue2.yaml

Thank you for your help and this great project of yours.
